spec/models/report_spec.rb

```diff
@@ -0,0 +1,121 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe Report, type: :model do
+  describe '#editable' do
+    let(:user) { User.create!(**params) }
+    let(:params) do
+      {
+        name: 'alice',
+        email: 'alice@example.com',
+        password: '123456',
+        password_confirmation: '123456'
+      }
+    end
+    let(:another_user) { User.create!(**another_params) }
+    let(:another_params) do
+      {
+        name: 'bob',
+        email: 'bob@example.com',
+        password: '123456',
+        password_confirmation: '123456'
+      }
+    end
+    let(:report) { Report.create!(**report_params) }
+    let(:report_params) do
+      {
+        user_id: user.id,
+        content: 'content',
+        title: 'title'
+      }
+    end
+    context 'user have report' do
+      it 'is editable' do
+        expect(report.editable?(user)).to be_truthy
+      end
+    end
+    context 'user don\'t have report' do
+      it 'is not editable' do
+        expect(report.editable?(another_user)).to be_falsey
+      end
+    end
+  end
+  before do
+  end
+  describe '#created_on' do
+    let(:user) { User.create!(**params) }
+    let(:params) do
+      {
+        name: 'alice',
+        email: 'alice@example.com',
+        password: '123456',
+        password_confirmation: '123456'
+      }
+    end
+    let(:report) { Report.create!(**report_params) }
+    let(:report_params) do
+      {
+        user_id: user.id,
+        content: 'content',
+        title: 'title',
+        created_at: 'Tue, 24 Jun 2024 14:33:07.146486000 JST +09:00'
+      }
+    end
+    it 'is created' do
+      expect(report.created_on).to eq Date.new(2024, 6, 24)
```

Comment:
```suggestion
      report.created_at = ....
      expect(report.created_on).to eq Date.new(2024, 6, 24)
```

このタイミングでテスト用にcreated_atをセットするのがいいかも。
（原則は固定値を使わない方がいいですが、ここは例外として）